.github/workflows/build-and-run.yml

```
@@ -0,0 +1,109 @@
+name: Build-And-Run Tests
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|ionosphere|background|pybind.unwrap.phass
+
+    defaults:
+      run:
+        shell: bash -l {0}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - label: Linux
+            runner: ubuntu-latest
+            ctest_exclude: -E ".*($ALL_EXCLUDE)"
+          - runner: macos-latest
+            label: macOS
+            ctest_exclude: -E ".*($ALL_EXCLUDE|raster|unwrap.snaphu)"
```

There's actually a few different tests with "raster" in their name, and I think I only expect one of them to fail on macOS.

Let's see if we can make this pattern a little more specific in order to avoid unnecessarily excluding some tests

```suggestion
            ctest_exclude: -E ".*($ALL_EXCLUDE|io\.gdal\.raster|unwrap.snaphu)"
```